File: Conditional_Routing/nodejs/custom_headers.js
const TAG = 'x-replace';

async function f(event) {
    const request = event.request;
    const headers = request.headers;

    let prefix = '';
    // 多值取第一个
    if (headers[TAG]) {
        prefix = '/' + headers[TAG].split(',')[0];
    }

    // 更新 uri
    request.uri = prefix + request.uri;

    // 自动回源
    return request;

}

exports.handler = f;

File: Conditional_Routing/python/custom_headers.py
TAG = 'x-replace'


async def handler(event):
    request = event.request
    headers = request.headers

    prefix = ''
    # 多值取第一个
    if TAG in headers:
        prefix = '/' + headers.get(TAG)

    # 更新 uri
    request.uri = prefix + request.uri

    # runtime 自动回源
    return Request(request=request)
